samples/MegaDrone/src/main/cpp/native-lib.cpp

```diff
@@ -21,20 +21,19 @@
 #include "MegaDroneEngine.h"
 
 
+namespace {
+
 std::vector<int> convertJavaArrayToVector(JNIEnv *env, jintArray intArray){
 
-    std::vector<int> v;
-    jsize length = env->GetArrayLength(intArray);
+    jsize length;
+    std::unique_ptr<jint[]> elements =
+        CHECK_JNI(FATAL, env).GetIntArrayElements(intArray, &length);
```

Comment:
GetIntArrayElements() does not take a length pointer. It takes a pointer to a boolean that indicates whether a copy of the data was made.

Also we need to balance this with ReleaseIntArrayElements so that the Java resource is free up.